src/CommonDBTM.php

```diff
@@ -5771,34 +5788,86 @@ private function assetBusinessRules($condition)
+                $existing_data['_groups_id_tech'],
+                is_array($this->fields['groups_id_tech'] ?? null) ? $this->fields['groups_id_tech'] : []
+            );
+
+            // Execute all defined rules with the specified condition (add or update)
             $output = $ruleasset->processAllRules($input, [], [], [
                 'condition' => $condition,
             ]);
 
-            // If at least one rule has matched
+            // Apply rule results to input data
+            // If at least one rule has been applied
             if (isset($output['_rule_process'])) {
-                foreach ($output as $key => $value) {
-                    if ($key === '_rule_process' || $key === '_no_rule_matches') {
+                foreach ($output as $field => $value) {
+                    // Skip internal rule processing keys
+                    if ($field === '_rule_process' || $field === '_no_rule_matches') {
                         continue;
                     }
-                    // Add the rule output to the input array
-                    $this->input[$key] = $value;
-                }
+
+                    $value_processed = false;
+
+                    // Add the rule result to the object's input data
+                    if (is_array($value) && $value !== []) {
+                        // determine $preserve_user_input by processing the first item, the value is the same on each entry
+                        // but just in case this is not the case anymore a check is processed
+                        // value can be an array if multiples values are set and can be an array because of appendtoarray & appendtoarrayfield are defined in action (for group fields, _groups_id , _groups_id_tech)
+                        // @see \RuleAsset::getActions()
+                        $_rule_action_field_exists = array_reduce(
+                            $value,
+                            static fn($result, $v) => $result && is_array($v) && array_key_exists('_rule_action', $v),
+                            true
+                        );
+                        if ($_rule_action_field_exists) {
+
+                            $preserve_user_input = $value[0]['_rule_action'] === 'append';
+                            if ($preserve_user_input && !isset($existing_data[$field])) {
+                                throw new LogicException('Implemetation error: action was defined with appendtoarray but the value not backed up before rule processing');
+                            }
+                            // use value before rule processing
+                            $this->input[$field] = $preserve_user_input ? $existing_data[$field] : [];
+                            foreach ($value as $_value) {
+                                $this->input[$field][] = $_value[$field];
+                            }
+                            $value_processed = true;
+                        }
+                    }
```

Comment:
If possible , the related logic should be located in the rules classes. However, I do not know where the logic used for ticket actors is located.